client.go

```diff
@@ -296,45 +321,36 @@ type WSClient struct {
 // To specify a callback function to process the notification
 //
 //   WithWebsocketCallBack(p PullUnreadMessages)
-//
-// To set the max number of notifications that user want to receive ( used in test only)
-//
-// After receiving enough notifications, the socket stop and close
-//
-//   WithMaxNotified(n uint64)
-func NewWSClient(opts ...SPVConfigFunc) *WSClient {
+func NewWSClient(opts ...SPVConfigFunc) (*WSClient, error) {
 	// Start with the defaults then overwrite config with any set by user
 	cfg := defaultSPVConfig()
 	for _, opt := range opts {
 		opt(cfg)
 	}
 
-	if cfg.procces == nil {
-		cfg.procces = func(t int, msg []byte, err error) error {
-			return nil
-		}
+	ws := &WSClient{
+		cfg:   cfg,
+		ws:    nil,
+		close: make(chan bool),
 	}
 
-	return &WSClient{
-		cfg:        cfg,
-		ws:         nil,
-		nbNotified: 0,
+	err := ws.connectServer()
+
+	if err != nil {
+		return nil, err
 	}
+
+	return ws, nil
 }
 
 // urlPath return the path part of the connection URL
 func (c *WSClient) urlPath() string {
 	return fmt.Sprintf("/api/%s/channel/%s/notify", c.cfg.version, c.cfg.channelID)
 }
 
-// NbNotified return the number of processed messages
-func (c *WSClient) NbNotified() uint64 {
-	return c.nbNotified
-}
-
-// Run establishes the connection and start listening the notification stream
-// process the notification if a callback is provided
-func (c *WSClient) Run() error {
+// connectServer establish the connection to the server
+// Return error if any
+func (c *WSClient) connectServer() error {
 	u := url.URL{
 		Scheme: "wss",
 		Host:   "localhost:5010",
```

Comment:
Should these values should be read from the config?

Reply:
oh nice one. I'll fix that